Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -532,4 +604,170 @@ lemma IsIntegralCurve.periodic_xor_injective [BoundarylessManifold I M]
       rw [abs_of_nonneg hab]
       exact hγ.periodic_of_eq hv heq
 
+section UniformTime
+
+lemma exists_isIntegralCurveOn_Ioo_eqOn [BoundarylessManifold I M]
```

Comment:
This is a strange-looking lemma to me. Are you sure you need it?

If you do need then, I would state it as:
```lean
lemma eqOn_of_isIntegralCurveOn_Ioo [BoundarylessManifold I M]
    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M))) {x : M}
    (γ : (a : ℝ) → ℝ → M) (hγx : ∀ a, γ a 0 = x) (hγ : ∀ a, IsIntegralCurveOn (γ a) v (Ioo (-a) a))
    {a a' : ℝ} (hpos : 0 < a') (hle : a' ≤ a) :
    EqOn (γ a') (γ a) (Ioo (-a') a') := by
```

Reply:
Yeah, this is my first time dealing with `Classical.choose`, so I'm curious if there's a better way to do this. The reason I need this lemma is that I'm trying to build a global integral curve from integral curves defined on nested intervals. I need the longer integral curve to agree with the shorter integral curve, wherever the latter is defined.